packages/cli-repl/src/cli-repl.ts

```diff
@@ -391,7 +391,15 @@ export class CliRepl implements MongoshIOProvider {
         axiosRetryConfig: { retries: 0 }
       } as any /* axiosConfig and axiosRetryConfig are existing options, but don't have type definitions */
     );
-    this.toggleableAnalytics = new ToggleableAnalytics(this.segmentAnalytics);
+    this.toggleableAnalytics = new ToggleableAnalytics(
+      new ThrottledAnalytics({
+        target: this.segmentAnalytics,
+        throttle: {
+          rate: 30,
+          metadataPath: this.shellHomeDirectory.paths.shellLocalDataPath
```

Comment:
Is this an okay path to use here or maybe we want some special `tmpPath` provided from `shellHomeDirectory` for this? I'm open to suggestions

Reply:
I think this is fine, yes 👍